src/core/public/workspace/workspaces_service.ts

```diff
@@ -74,11 +76,8 @@ export class WorkspacesService implements CoreService<WorkspacesSetup, Workspace
             /**
              * Current workspace is stale
              */
-            this.currentWorkspaceId$.error({
-              reason: WORKSPACE_ERROR.WORKSPACE_IS_STALE,
-            });
-            this.currentWorkspace$.error({
-              reason: WORKSPACE_ERROR.WORKSPACE_IS_STALE,
+            this.workspaceError$.error({
```

Comment:
As there is already an error observable, should we just
```
workspaceError$.next('{some error reason}')
```
?

Reply:
I think we still need this to be a error because it will be easier for us to do the `workspaceError$.hasError` check. I think for some security reason use `workspaceError$.hasError.value` may cause some issue.

Reply:
Done